[quantization] Show a warning that not all dir members of quantized work.

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#27339 [quantization] Show a warning that not all dir members of quantized work.**

This PR just shows a warning message.
Eventually we will show a correct __dir__

Differential Revision: [D17751333](https://our.internmc.facebook.com/intern/diff/D17751333)

Commit:
[quantization] Show a warning that not all dir members of quantized work.

This PR just shows a warning message.
Eventually we will show a correct __dir__

raghuramank100 commented on torch/tensor.py:
```diff
@@ -414,6 +414,10 @@ def __hash__(self):
+        if self.is_quantized:
+            warnings.warn('dir() of a quantized tensor is possibly wrong. '
+                          'Not all members are supported (yet). '
+                          'Use at your own disgression!')
```
Please change to:
'Only a small subset of methods are supported for quantized tensors. '

Commit:
Update on "[quantization] Show a warning that not all dir members of quantized work."


This PR just shows a warning message.
Eventually we will show a correct __dir__

Differential Revision: [D17751333](https://our.internmc.facebook.com/intern/diff/D17751333)

Commit:
Update on "[quantization] Show a warning that not all dir members of quantized work."


This PR just shows a warning message.
Eventually we will show a correct __dir__

Differential Revision: [D17751333](https://our.internmc.facebook.com/intern/diff/D17751333)
